core/cmd/client.go

```diff
@@ -75,24 +80,66 @@ type ChainlinkAppFactory struct{}
 
 // NewApplication returns a new instance of the node with the given config.
 func (n ChainlinkAppFactory) NewApplication(cfg config.GeneralConfig) (chainlink.Application, error) {
-	advisoryLocker := postgres.NewAdvisoryLock(cfg.DatabaseURL())
 	shutdownSignal := gracefulpanic.NewSignal()
-	// TODO: Remove store entirely
-	// https://app.clubhouse.io/chainlinklabs/story/12980/remove-store-object-entirely
-	store, err := strpkg.NewStore(cfg, advisoryLocker, shutdownSignal)
+	uri := cfg.DatabaseURL()
+	dialect := cfg.GetDatabaseDialectConfiguredOrDefault()
+	db, gormDB, err := postgres.NewConnection(uri.String(), string(dialect), cfg, shutdownSignal)
 	if err != nil {
 		return nil, err
 	}
-	db := store.DB
-	sqlxDB := postgres.UnwrapGormDB(db)
-	cfg.SetDB(db)
-	keyStore := keystore.New(db, utils.GetScryptParams(cfg))
+	cfg.SetDB(gormDB)
+
+	// Set up the versioning ORM
+	verORM := versioning.NewORM(db)
+
+	// Set up periodic backup
+	if cfg.DatabaseBackupMode() != config.DatabaseBackupModeNone {
+		var version *versioning.NodeVersion
+		var versionString string
+
+		version, err = verORM.FindLatestNodeVersion()
+		if err != nil {
+			if errors.Is(err, sql.ErrNoRows) {
+				logger.Default.Debugf("Failed to find any node version in the DB: %w", err)
+			} else if strings.Contains(err.Error(), "relation \"node_versions\" does not exist") {
+				logger.Default.Debugf("Failed to find any node version in the DB, the node_versions table does not exist yet: %w", err)
+			} else {
+				return nil, errors.Wrap(err, "initializeORM#FindLatestNodeVersion")
+			}
+		}
+
+		if version != nil {
+			versionString = version.Version
+		}
+
+		databaseBackup := periodicbackup.NewDatabaseBackup(cfg, logger.Default)
+		databaseBackup.RunBackupGracefully(versionString)
+	}
+
+	// Migrate the database
+	if cfg.MigrateDatabase() {
```

Comment:
Feels really hokey to do all these mutative/destructive actions in `NewApplication`, but I guess it always did this and it was simply hidden